finagle-http/src/main/scala/com/twitter/finagle/http/filter/ComplianceFilter.scala

```diff
@@ -0,0 +1,62 @@
+package com.twitter.finagle.http.filter
+
+import com.twitter.finagle.http.{Fields, Request, Response, Status}
+import com.twitter.finagle._
+import com.twitter.logging.Logger
+import com.twitter.util.Future
+import org.jboss.netty.buffer.ChannelBuffers
+
+/**
+ * Modify a content and header field filter.
+ *
+ * @see [[https://tools.ietf.org/html/rfc2616 "Hypertext Transfer Protocol -- HTTP/1.1"]]
+ * @see [[https://tools.ietf.org/html/rfc7230 "Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing"]]
+ * section-3.3.2 on `Content-Length` for more information.
+ */
+private[finagle] class ComplianceFilter[Req <: Request] extends SimpleFilter[Req, Response] {
+  import ComplianceFilter._
+  val logger = Logger(getClass)
+
+  def apply(request: Req, service: Service[Req, Response]): Future[Response] = {
+    service(request).map { rep =>
+      if (mustNotIncludeContent(rep.status)) {
+        if (rep.getContent() != ChannelBuffers.EMPTY_BUFFER) {
+          rep.clearContent()
+          logger.warning(s"Response with a status code of ${rep.statusCode} must not have a body-message thus the content has been removed.")
+        }
+
+        if (rep.status == Status.NoContent && rep.contentLength.isDefined) {
+          rep.headerMap.remove(Fields.ContentLength)
+          logger.warning(s"Response with a status code of ${rep.statusCode} must not have a Content-Length header field thus the field has been removed.")
+        }
+
+        //Make sure a Content-Length header field is set with a value of 0 if it is explicitly sent for a 304 response
```

Comment:
Oops, I think I was wrong. As RFC 7230 section3.3.2 says, if we do want to send a Content-Length header field, we should return the Content-Length value of the cached content which has been returned to a client in a 200 response. I thought a value of Content-Length header field was always 0 as the content of a 304 response is always empty...
The following is the citation from the section;
`A server MAY send a Content-Length header field in a 304 (Not Modified) response to a conditional GET request (Section 4.1 of [RFC7232]); a server MUST NOT send Content-Length in such a response unless its field-value equals the decimal number of octets that would have been sent in the payload body of a 200 (OK) response to the same request.`